Ryujinx.HLE/HOS/Services/Mii/Types/StoreData.cs

```diff
@@ -1,7 +1,5 @@
-﻿using LibHac.Common;
-using Ryujinx.HLE.Utilities;
+﻿using Ryujinx.Common.Utilities;
```

Comment:
Switch back to the other `SpanHelpers` class to get proper lifetime tracking
```suggestion
using LibHac.Common;
```

Reply:
Ryujinx does the exact same with other struct members elsewhere in the code; this just matches these two cases to the rest of the codebase.

Since these two uses aren’t really specifically related to LibHac at all but just happened to use it’s helper I would argue if there’s an issue there would need to be a larger fix of the Ryujinx SpanHelpers outside of this PR, and aligning these uses onto the same SpanHelpers would suffice for now.

Reply:
In general we don't want to add a dependency to Libhac in the code if we don't need to, as if we want to use the code in a separate project or standalone we don't want to depend on Libhac just for minor utility methods.

Reply:
The `SpanHelpers` class in Ryujinx is (or was) just a copy of the class in Libhac, so it matter too much which one is used outside of possible naming collisions. I only suggested using the one in Libhac because it doesn't have the incorrect `scoped` parameters, and fixing the one in Ryujinx would probably require properly annotating other functions throughout Ryujinx

Reply:
I think the right thing to do here is fixing the one on Ryujinx, or removing the one on Ryujinx and using the Libhac method everywhere. In both cases it would require all existing uses of those methods. My preference would be of course keeping and fixing the one on Ryujinx due to the reasons I mentioned before. In any case, if this is a existing issue and the code is working as is, I think it would be fine to fix in a separate PR.

Reply:
The code in question in the PR is safe as it currently is. It's just bypassing the compiler's lifetime tracking so it's up to the developer to make sure that the code's safe. It should be fine to commit it as-is and do the fix in a separate PR.

Reply:
I would really prefer that those helpers be from one dependency.

However, I do not want to bring LibHac dependency on non HLE parts, this part would need to be spitted out of it.

Not to mention it could be beneficial for the .NET ecosystem in general.

In the meantime, @Steveice10 can you revert the changes here and add a TODO comment stating the lifetime issue?

Reply:
> I would really prefer that those helpers be from one dependency.
> 
> However, I do not want to bring LibHac dependency on non HLE parts, this part would need to be spitted out of it.
> 
> Not to mention it could be beneficial for the .NET ecosystem in general.
> 
> In the meantime, @Steveice10 can you revert the changes here and add a TODO comment stating the lifetime issue?

This change is actually removing the LibHac dependency. Without this change it won't compile because of LibHac changes.

Reply:
urgh right

> > I would really prefer that those helpers be from one dependency.
> > However, I do not want to bring LibHac dependency on non HLE parts, this part would need to be spitted out of it.
> > Not to mention it could be beneficial for the .NET ecosystem in general.
> > In the meantime, @Steveice10 can you revert the changes here and add a TODO comment stating the lifetime issue?
> 
> This change is actually removing the LibHac dependency. Without this change it won't compile because of LibHac changes.

Okay, I guess it's fine then.

It would be nice to open an issue for that @Thealexbarney, I will try to handle it when I get some times (or someone can look into that).